modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/APPMPublisherRestClient.java

```diff
@@ -0,0 +1,382 @@
+            AppCreateRequest appRequest = new AppCreateRequest(appName, context, appVersion,
+                                                               trackingCode);
+            appRequest.setUriTemplatePolicyGroupId0(policyId);
+            appRequest.setUriTemplatePolicyGroupId1(policyId);
+            appRequest.setUriTemplatePolicyGroupId2(policyId);
+            appRequest.setUriTemplatePolicyGroupId3(policyId);
+            appRequest.setUriTemplatePolicyGroupId4(policyId);
+            appRequest.setUriTemplatePolicyGroupIds(policyGroupId);
+
+            HttpResponse appCreateResponse = createApp(appRequest);
+            if (appCreateResponse.getResponseCode() == 200) {
+                HttpResponse addSsoProviderResponse = addSsoProvider(appRequest);
+                if (addSsoProviderResponse.getResponseCode() == 200) {
+                    return appCreateResponse;
+                } else {
+                    throw new Exception("Error occurred while service provider creating! "
+                                                + appCreateResponse.getData());
+                }
+            } else {
+                throw new Exception("Error occurred while new web app creating! "
+                                            + appCreateResponse.getData());
+            }
+        } else {
+            throw new Exception("Error occurred while policy adding! " + httpResponseData);
+        }
+    }
+
+    /**
+     * Add policy group before creating the application.
+     *
+     * @param policyDesc String.
+     * @return httpResponse.
+     * @throws Exception on errors.
+     */
+    private HttpResponse addPoicyGroup(String policyDesc) throws Exception {
+        requestHeaders.put(AppmTestConstants.CONTENT_TYPE, "application/x-www-form-urlencoded");
+
+        String payload = "policyGroupName=Default&throttlingTier=Unlimited&userRoles"
+                + "=&anonymousAccessToUrlPattern=false&objPartialMappings"
+                + "=[]&policyGroupDesc="
+                + policyDesc;
+
+        HttpResponse response = HttpRequestUtil.doPost(new URL(backEndUrl
+                                                       + AppmTestConstants.PubliserRestApis
+                                                                                  .ADD_POLICY_GROUP),
+                                                       payload, requestHeaders);
+        return response;
+    }
+
+    /**
+     * creating web application.
+     *
+     * @param appRequest - to create the payload.
+     * @return response HttpResponse.
+     * @throws Exception on errors.
+     */
+    private HttpResponse createApp(AppCreateRequest appRequest) throws Exception {
+        String payload = appRequest.generateRequestParameters();
+        String roles = appRequest.getRoles();
+        this.requestHeaders.put(AppmTestConstants.CONTENT_TYPE,
+                                "application/x-www-form-urlencoded");
+        HttpResponse response =
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/commit/f021f86c3c398dddeeca17e225a37e42f7f7033b